tests/integration_tests/model_tests.py

```diff
@@ -578,6 +580,25 @@ def test_data_for_slices_with_query_context(self):
             "state",
         }
 
+    @pytest.mark.usefixtures("load_birth_names_dashboard_with_slices")
+    def test_data_for_slices_with_adhoc_column(self):
+        # should perform sqla.model.BaseDatasource.data_for_slices() with adhoc
+        # column and legacy chart
+        tbl = self.get_table(name="birth_names")
+        slc = (
+            metadata_db.session.query(Slice)
+            .filter_by(
+                datasource_id=tbl.id, datasource_type=tbl.type, slice_name="Boys",
+            )
+            .first()
+        )
```

Comment:
Can we create a new slice instead to reduce dependencies on other tests/data?

Reply:
Thanks, independent unit testing is necessary.

Reply:
done.